### PR TITLE
Updates to SQLCalls and finalized Profile Servlet

### PR DESCRIPTION
**Pull request by camshannon**
Made some changes to SQLCalls Profile section and finished up the profile servlet. Now ready for testing.

### CODE REVIEW

File: sql/DBDeply.sql [hunk 1]
```diff
@@ -10,8 +10,8 @@ create table Users(
     Username VARCHAR(100) NOT NULL,
     Password VARCHAR(20) NOT NULL,
     City VARCHAR(1000),
-    Stars DOUBLE NOT NULL,
-    Verified BOOLEAN NOT NULL,
+    Stars DOUBLE,
+    Verified BOOLEAN,
```
Comment (camshannon) [inline sql/DBDeply.sql:14]: Decided these should be NULL and use IFNULL() in SELECT statements.

File: src/models/Profile.java [hunk 2]
```diff
@@ -12,12 +12,12 @@
 	private ArrayList<Review> reviews;
 	
 	public Profile(String name,
-				   String city,
-				   double stars,
-				   boolean verified,
-				   String handicap,
-				   String picture,
-				   ArrayList<Review> reviews) {
+			String city,
```
Comment (camshannon) [inline src/models/Profile.java:15]: Visual changes here.

File: src/models/Review.java [hunk 3]
```diff
@@ -6,15 +6,23 @@
 	private String userName;
 	private int upvotes;
 	private int downvotes;
-	private Location location;
+	private String locationName;
+	private String image;
 	
-	public Review(String title, String body, String userName, int upvotes, int downvotes, Location location) {
+	public Review(String title,
```
Comment (camshannon) [inline src/models/Review.java:12]: More visual changes.

File: src/models/Review.java [hunk 4]
```diff
@@ -6,15 +6,23 @@
...
+	private String locationName;
+	private String image;
 	
-	public Review(String title, String body, String userName, int upvotes, int downvotes, Location location) {
+	public Review(String title, 
+			String body, 
+			String userName, 
+			int upvotes, 
+			int downvotes, 
+			String locationName,
```
Comment (camshannon) [inline src/models/Review.java:17]: I changed the Review class to only have a location name, not an entire location object. It seems to me that if we need the entire location object, we can just make a request to get the location data.

File: src/models/SQLCalls.java [hunk 5]
```diff
@@ -193,32 +193,44 @@ public void leaveRating(String locationID, String userID, String elevatorRating,
 // --------------------------------- For Profile.java ---------------------------------
 	
 	// sets a user's city
-	public void setCity(String city, String userID) {
+	public String setCity(String username, String city) {
```
Comment (camshannon) [inline src/models/SQLCalls.java:196]: I changed the parameters of the profile methods, so the frontend will need usernames, not userIDs, to make changes to User table.

File: src/models/SQLCalls.java [hunk 6]
```diff
@@ -193,32 +193,44 @@ public void leaveRating(String locationID, String userID, String elevatorRating,
...
 		try {
 			Statement st = conn.createStatement();
-			st.executeQuery("UPDATE Users SET City='" + city + "' WHERE UserID='" + userID + "'");
+			st.executeQuery("UPDATE Users SET City='" + city + "' WHERE Username='" + username + "'");
 		} catch(SQLException e) {
 			System.out.println("SQLException in setCity: " + e.getMessage());
+			return "SQLException in setCity: " + e.getMessage();
 		}
+		return "success";
```
Comment (camshannon) [inline src/models/SQLCalls.java:204]: Added return types so the request will either see success or the exception method if an exception has occurred.

**Comment by cvturner** [comment]
These all look good to me so far if we want to merge

**Pull request merged**